src/h5web/vis-packs/core/scales.ts

```diff
@@ -3,47 +3,106 @@ import {
   scaleLog,
   scaleSymlog,
   scaleSqrt,
-  PickScaleConfig,
+  scalePower,
 } from '@visx/scale';
 import type { IconType } from 'react-icons/lib';
-import type { AxisScale } from './models';
+import type { AxisScale, GammaScaleConfig, VisxScaleConfig } from './models';
 import { ScaleType } from './models';
-import { MdSort, MdFilterList } from 'react-icons/md';
+import { MdSort, MdFilterList, MdFlare } from 'react-icons/md';
 import SqrtIcon from '../../toolbar/controls/ScaleSelector/SqrtIcon';
 import MdGraphicEqRotated from '../../toolbar/controls/ScaleSelector/MdGraphicEqRotated';
+import type { NumberValue, ScalePower } from 'd3-scale';
 
 interface H5WebScale {
-  createScale: (
-    config: Omit<PickScaleConfig<ScaleType, number>, 'type'>
-  ) => AxisScale;
   Icon: IconType;
   label: string;
   validMin: number;
 }
 
 export const H5WEB_SCALES: Record<ScaleType, H5WebScale> = {
   [ScaleType.Linear]: {
-    createScale: (config) => scaleLinear<number>(config),
     Icon: MdSort,
     label: 'Linear',
     validMin: -Infinity,
   },
   [ScaleType.Log]: {
-    createScale: (config) => scaleLog<number>(config),
     Icon: MdFilterList,
     label: 'Log',
     validMin: Number.MIN_VALUE,
   },
   [ScaleType.SymLog]: {
-    createScale: (config) => scaleSymlog<number>(config),
     Icon: MdGraphicEqRotated,
     label: 'SymLog',
     validMin: -Infinity,
   },
   [ScaleType.Sqrt]: {
-    createScale: (config) => scaleSqrt<number>(config),
     Icon: SqrtIcon,
     label: 'Square root',
     validMin: 0,
   },
+  [ScaleType.Gamma]: {
+    Icon: MdFlare,
+    label: 'Gamma',
+    validMin: -Infinity,
+  },
 };
+
+export function gammaScale(
+  config: GammaScaleConfig
+): ScalePower<number, number> {
+  const linScale = scaleLinear<number>({ ...config, range: [0, 1] });
+  const powScale = scalePower<number>({
+    ...config,
+    domain: [0, 1],
+  });
+
+  function gamma(value: NumberValue): number {
+    return powScale(linScale(value));
+  }
+
+  const result = scalePower<number>(config);
+  return Object.assign(gamma, {
+    invert: (val: number) => linScale.invert(powScale.invert(val)),
+    domain: result.domain,
+    range: result.range,
+    rangeRound: result.rangeRound,
+    clamp: result.clamp,
+    unknown: result.unknown,
+    interpolate: result.interpolate, // To be fixed ?
```

Comment:
yep 😄 - right now it doesn't work with a domain other than `[0, 1]`, I assume?

Reply:
Good call, I will extend the tests to see if this is true